docs/basic-tools.md

```diff
@@ -333,8 +333,8 @@ def test_colinearity( points ):
 ```
 
 Before moving on to the next section, it is worth pausing to note that all ingredients are in place for *finite extension fields*, or simply *extension fields*. A finite field is simply a set equipped with addition and multiplication operators that behave according to high school algebra rules, *e.g.* every nonzero element has an inverse, or no two nonzero elements multiplied give zero. There are two ways to obtain them:
- 1. Start with the set of integers, and reduce the result of any addition or multiplication modulo a given prime number $p$.
- 2. Start with the set of polynomials over a finite field, and reduce the result of any addition or multiplication modulo a given *irreducible polynomial* $p(X)$. A polynomial is *irreducible* when it cannot be decomposed as the product of two smaller polynomials, analogously to prime numbers.
+ 1. Start with the set of integers, and reduce the result of any addition or multiplication modulo by a given prime number $p$.
+ 2. Start with the set of polynomials over a finite field, and reduce the result of any addition or multiplication modulo by a given *irreducible polynomial* $p(X)$. A polynomial is *irreducible* when it cannot be decomposed as the product of two smaller polynomials, analogously to prime numbers.
```

Comment:
```suggestion
 2. Start with the set of polynomials over a finite field, and reduce the result of any addition or multiplication modulo a given *irreducible polynomial* $p(X)$. A polynomial is *irreducible* when it cannot be decomposed as the product of two smaller polynomials, analogously to prime numbers.
```

The phrase "reduce the result modulo a prime number" is correct; no "by" needed.